app/components/app_session_overview_tallies_component.rb

```diff
@@ -0,0 +1,155 @@
+  def initialize(session)
+    @session = session
+    @patient_ids = session.patients.pluck(:id)
+    @academic_year = session.academic_year
+  end
+
+  attr_reader :session, :patient_ids, :academic_year
+
+  delegate :govuk_table, to: :helpers
+  delegate :programmes, to: :session
+
+  def tally_cards_for_programme(programme)
+    [
+      {
+        heading: "Eligible cohort",
+        colour: "blue",
+        count: eligible_for_vaccination_count(programme).to_s,
+        link_to: nil
+      },
+      {
+        heading: "Vaccinated",
+        colour: "green",
+        count: vaccinated_count(programme).to_s,
+        link_to:
+          session_patients_path(
+            session,
+            vaccination_status: "vaccinated",
+            programme_types: [programme.type]
+          )
+      },
+      {
+        heading: "Could not vaccinate",
+        colour: "red",
+        count: could_not_vaccinate_count(programme).to_s,
+        link_to:
+          session_patients_path(
+            session,
+            vaccination_status: "could_not_vaccinate",
+            programme_types: [programme.type]
+          )
+      },
+      {
+        heading: "No outcome",
+        colour: "grey",
+        count: no_outcome_count(programme).to_s,
+        link_to:
+          session_patients_path(
+            session,
+            vaccination_status: "none_yet",
+            programme_types: [programme.type]
+          )
+      }
+    ]
+  end
+
+  private
+
+  def eligible_for_vaccination_count(programme)
+    @eligible_for_vaccination_count ||= {}
+    @eligible_for_vaccination_count[
+      programme.id
+    ] ||= patients_in_programme_cohort_count(programme) -
+      previously_vaccinated_count(programme) -
+      vaccinated_at_different_locations_count(programme)
+  end
+
+  def vaccinated_count(programme)
+    [
+      0,
+      administered_vaccination_count(programme) -
+        previously_vaccinated_count(programme)
+    ].max
+  end
+
+  def could_not_vaccinate_count(programme)
+    patients_for_programme(programme).has_vaccination_status(
+      :could_not_vaccinate,
+      programme:,
+      academic_year:
+    ).count
+  end
+
+  def no_outcome_count(programme)
+    patients_for_programme(programme).has_vaccination_status(
+      :none_yet,
+      programme:,
+      academic_year:
+    ).count
+  end
+
+  def patients_in_programme_cohort_count(programme)
+    @patients_in_programme_cohort ||= {}
+    @patients_in_programme_cohort[programme.id] ||= patients_for_programme(
+      programme
+    ).appear_in_programmes(programme, academic_year:).count
+  end
+
+  def previously_vaccinated_count(programme)
+    return 0 if programme.seasonal?
+
+    @previously_vaccinated_count ||= {}
+    @previously_vaccinated_count[programme.id] ||= patients_for_programme(
+      programme
+    ).has_vaccination_status(
+      :vaccinated,
+      programme:,
+      academic_year: ...academic_year
```

Comment:
```suggestion
      academic_year: academic_year - 1
```

I think this should work in the same way as even if a patient is vaccinated many years ago, they'll be vaccinated last year.